### DISCUSSION TITLE
Lazer more susceptible to dual monitor latency than stable?

### DISCUSSION THREAD

**Discussion by TheFantasticLoki** [ppy/osu#16970] [Q&A] [answered]
I am opening this discussion to ask the community if they are experiencing the same issue as I am.
The issue I am experiencing may not be very noticeable to most but to those that can it's a big issue. That being increased perceived latency with a program such as discord video chat or a youtube video being displayed on a dual monitor while playing osu.
I originally thought this was lazer itself just feeling like it had more latency that stable, but after closing all other windows open on my second monitor AND pausing wallpaper engine, the input delay was gone. 
I do not notice this much of a hit in input delay on stable compared to lazer, as on stable I can play multiplayer with discord open on my second monitor with everyone's streams open and barely take a hit in perceived delay. But on lazer just having wallpaper engine running the wallpaper on my second monitor takes my cursor from having no delay to falling behind by half my screen sometimes in 7-8* jump maps and I can actually feel the delay between what I see on screen vs where my pen is on my tablet.

I am opening this discussion to ask the community if they have also noticed the increase in perceived delay with something running on their second monitor and hope we can remedy the issue.

**Comment by peppy** [comment]
You need to run lazer and any osu! using exclusive fullscreen. If it's not, it will be fucked by anything else running on the system.

**Reply by TheFantasticLoki** [reply]
I do run in exclusive fullscreen mode I always do because of the input delay hit it takes in any game.

**Reply by peppy** [reply]
This means that something else on your PC is likely handling the input. Your probably also have your mouse running at 10,000hz for no good reason.

Open the frame graphs in lazer (ctrl+f11 twice) and look at the input thread. If you see grey there, it means something else on your system is using the window loop thread for processing and causing delays in lazer (and any other app).

**Reply by TheFantasticLoki** [reply]
I have my mouse regrettably set to 250hz due to call of duty black ops 2 not handling high polling rate well so that's not it.

and on the frame graphs I only see occasional little spikes once every like 2 minutes on the input graph even tho I can still feel the delay with discord on my second monitor. This has been an established problem that effects all games on windows it's just a matter of how bad the problem is from app to app like how I don't notice any of these delay issues when on stable and I do on lazer.

**Reply by TheFantasticLoki** [reply]
Also I use tablet with OTD standalone and not the built in driver since I switch between stable and lazer a lot and I have raw input off on both lazer and stable since with tablet it only adds more processing delay.

**Reply by TheFantasticLoki** [reply]
This is usually not an input delay problem more so a frame time delay issue.

**Reply by peppy** [reply]
Please stop posting before I limit your ability to do so.

**Reply by TheFantasticLoki** [reply]
Sorry for posting again.... but I have to add this bit of information. With JUST discord running in osu!game VC with 1-2 streams on my second monitor input graph takes no hit but the Update graph goes from 680 fps avg during gameplay with nothing on the second monitor to 550 fps with discord on my second monitor.

Edit: Sliders are also causing the Update thread to lag a lot even with nothing open on second monitor and having auto play the map. 

Edit 2: Here is a good example map that really tanks the update thread, got me avg around 400 fps sometimes dipping into 300 fps territory: https://osu.ppy.sh/beatmapsets/1191248#osu/2482032

**Reply by peppy** [reply]
Your monitor doesn't run at 400 or 300 fps, so those numbers are still fine and should not be visible in any way.

**Reply by TheFantasticLoki** [reply]
if my monitor is 60hz and my game runs at 240hz it's still going to feel more responsive because of the decrease in frame delivery delay. It does indeed make a noticable impact and if you want a more in depth explanation please refer to this video as it does an excellent job of explaining "Frame Latency": https://youtu.be/hjWSRTYV8e0?t=109

**Reply by peppy** [reply]
for the sake of anyone else visiting this thread and being misinformed: the returns are diminishing and running 240fps or higher for a 60hz display is our recommendation. anything higher is roughly pointless and should not affect you in a visual way.

if your goal is to focus on the "second monitor causing performance issues" then please don't bring this up here. just close things on your second monitor.

good luck.

**Comment by pundice2** [comment]
Yeah, I actually noticed this too. The difference isn't huge and granted I have a pretty bad PC but I did notice some latency differences while having something open on my other monitor, and not. Generally this isn't as big of an issue with most other games but I noticed that while I play osu!lazer, it is a lot more significant.

**Comment by TheFantasticLoki** [comment, marked as answer]
For those who ARE having issue please disable g-sync or freesync if your monitor has it enabled as this did remove a LOT of the perceived delay for myself, I can now play just fine with very minimal delay while having Discord open on my second monitor with 6 livestreams running at once. Hopefully this does help you.

**Reply by peppy** [reply]
Please note that this is not the recommendation of the developers of this project, and I for one have no issues with gsync on (in fact it is as you would expect, smoother).
